### ISSUE TITLE
Header Modification Failure - Node 0.12

### ISSUE THREAD

**Issue by dstack** [caolan/quip#14]
```
_http_outgoing.js:386
throw new Error('Can\'t render headers after they are sent to the client.'
^
Error: Can't render headers after they are sent to the client.
at ServerResponse.OutgoingMessage._renderHeaders (_http_outgoing.js:386:11)
at ServerResponse.writeHead (_http_server.js:199:20)
at ServerResponse.res.write (C:\apps\swatchr\node_modules\quip\lib\quip.js:1
15:13)
at ServerResponse.OutgoingMessage.end (_http_outgoing.js:544:16)
at Array.write (C:\apps\swatchr\node_modules\connect\node_modules\finalhandl
er\index.js:126:9)
at listener (C:\apps\swatchr\node_modules\connect\node_modules\finalhandler\
node_modules\on-finished\index.js:164:15)
at onFinish (C:\apps\swatchr\node_modules\connect\node_modules\finalhandler\
node_modules\on-finished\index.js:95:5)
at callback (C:\apps\swatchr\node_modules\connect\node_modules\finalhandler\
node_modules\on-finished\node_modules\ee-first\index.js::10)
at IncomingMessage.onevent (C:\apps\swatchr\node_modules\connect\node_module
s\finalhandler\node_modules\on-finished\node_modules\ee-first\index.js:66:5)
at IncomingMessage.emit (events.js:104:17)
```

Specific test env was Node 0.12.2, using connect 3.3.5, on Windows 7 x64 and Windows 8x64.  Since this worked well in Node 0.10, I suspect node 0.12 is the culprit, specifically the http module.

**Comment by dstack** [comment]
Found the problem and the solution.  Since This module does not seem to receive recent active development, I will leave the info here for others that find the same problem.

Most recent version of connect was upgraded to use a module called "finalhandler".  This module uses meta information about the res to write the headers and end the response.  To fix quip to work with htis, change the res.write method.

Before:
    res.writeHead(statusCode, headers)

After:
    res.statusCode = statusCode;
    for(var header in headers){ res.setHeader(header, headers[header]); }

This modification allows quip and finalhandler to play nice together for new connect, but would probably not work with older connect (haven't tested yet).

**Comment by justrhysism** [comment]
+1

**Comment by ccorcos** [comment]
you should fork it!

**Comment by ccorcos** [comment]
https://github.com/caolan/quip/pull/15

**Comment by caolan** [comment]
Fixed by #15 and release as quip 0.1.3
